Palindrome.java

```diff
@@ -0,0 +1,115 @@
+package WEEK1;
+
+import static org.testng.Assert.assertEquals;
+
+//import java.util.Scanner;
+
+import org.junit.Test;
+
+public class Palindrome {
+	static int length = 0;
+	
+	public static boolean is_palindrome(Node first, Node second){
+		Node temp_f = first;
+		Node temp_s = second;
+		
+		while(temp_f!=null && temp_f.next!=second){
+			if(temp_f.data == temp_s.data){
+				temp_f = temp_f.next;
+				temp_s = temp_s.next;
+			}
+			else{
+				return false;
+			}
+		}
+		return true;
+	}
+	
+	public static Node reverse(Node start){
+		Node current = start;
+		Node prev = null;
+		Node next;
+		while(current!=null){
+			next = current.next;
+			current.next = prev;
+			prev = current;
+			current = next;
+		}
+		
+		start = prev;
+		return start;
+	}
+	
+	public static boolean palindrome(Node start){
```

Comment:
How would the execution look like if the linked list has one element?

Reply:
For one node, I am considering it as True, though not sure if it would be true or false. So that would make both second and first pointer to the same node and would return true.

Reply:
I have changed it to false. Realized that it should be false for a single element.